### PR TITLE
feat(panel): Add $animate.pin to the panel.

### PR DESCRIPTION
**Pull request by KarenParker**
Fixes: #7874.

@ErinCoughlan @jelbourn @gmoothart @DerekLouie Please review.

### CODE REVIEW

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -1065,6 +1065,12 @@ MdPanelRef.prototype._createPanel = function() {
             self._panelEl.addClass(self._config['panelClass']);
           }
 
+          // Panel may be outside the $rootElement, tell ngAnimate to animate
+          // regardless.
+          if ($animate.pin) {
```
Comment (ErinCoughlan) [inline src/components/panel/panel.js]: Don't you need to inject $animate? I didn't think it just existed.

Comment (KarenParker) [inline src/components/panel/panel.js]: Missed that from my other branch. Done.

File: src/components/panel/panel.js [hunk 2]
```diff
@@ -1065,6 +1068,12 @@ MdPanelRef.prototype._createPanel = function() {
             self._panelEl.addClass(self._config['panelClass']);
           }
 
+          // Panel may be outside the $rootElement, tell ngAnimate to animate
+          // regardless.
+          if (self.$animate.pin) {
```
Comment (ErinCoughlan) [inline src/components/panel/panel.js]: self._$animate

Comment (KarenParker) [inline src/components/panel/panel.js]: Done.

**Comment by ErinCoughlan** [comment]
LGTM

**Pull request merged**